keps/sig-node/3953-node-resource-hot-plug/README.md

```diff
@@ -0,0 +1,820 @@
+
+#### Story 1
+
+As a Kubernetes user, I want to allocate more resources (CPU, memory) to a node with existing specialized hardware or CPU Capabilities (for example:https://www.kernel.org/doc/html/v5.8/arm64/elf_hwcaps.html)
+so that additional workloads can leverage the hardware to be efficiently scheduled and run without manual intervention.
+
+#### Story 2
+
+As a Kubernetes Application Developer, I want the kernel to optimize system performance by making better use of local resources when a node is resized, so that my applications run faster with fewer disruptions. This is achieved when there are
+Fewer Context Switches: With more CPU cores and memory on a resized node, the kernel has a better chance to spread workloads out efficiently. This can reduce contention between processes, leading to fewer context switches (which can be costly in terms of CPU time) 
+and less process interference and also reduces latency.
+Better Memory Allocation: If the kernel has more memory available, it can allocate larger contiguous memory blocks, which can lead to better memory locality (i.e., keeping related data closer in physical memory), 
+reducing latency for applications that rely on large datasets, in the case of a database applications.
+
+#### Story 3
+
+As a Site Reliability Engineer (SRE), I want to reduce the operational complexity of managing multiple worker nodes, so that I can focus on fewer resources and simplify troubleshooting and monitoring.
+
+#### Story 4
+
+As a Cluster administrator, I want to resize a Kubernetes node dynamically, so that I can quickly hot plug resources without waiting for new nodes to join the cluster.
+
+#### Story 5
+
+As a Cluster administrator, I expect my existing workloads to function without having to undergo a disruption which is induced during capacity addition followed by a node/kubelet restart to
+detect the change in compute capacity, which can bring in additional complications.
+
+### Notes/Constraints/Caveats (Optional)
+
+### Risks and Mitigations
+
+- Change in OOMScoreAdjust value:
+    - The formula to calculate OOMScoreAdjust is `1000 - (1000*containerMemReq)/memoryCapacity`
+    - With change in memoryCapacity post up-scale, The existing OOMScoreAdjust may not be inline with the
+      actual OOMScoreAdjust for existing pods.
+        - This can be mitigated by recalculating the OOMScoreAdjust value for the existing pods. However, there can be an associated overhead for
+          recalculating the scores.
+- Change in Swap limit:
+    - The formula to calculate the swap limit is `<containerMemoryRequest>/<nodeTotalMemory>)*<totalPodsSwapAvailable>`
+    - With change in nodeTotalMemory and totalPodsSwapAvailable post up-scale, The existing swap limit may not be inline with the
+      actual swap limit for existing pods.
+        - This can be mitigated by recalculating the swap limit for the existing pods. However, there can be an associated overhead for
+          recalculating the scores.
```

Comment:
Which overhead are you referring to here?
IIUC there might be some overhead from the CRI side, but the limit calculation itself should be very very quick. AFAICT the CRI operation shouldn't be heavy as well, since it only writes a single value to a cgrpoupfs file per container.

Reply:
Yes, you are right. Just wanted to mention this overhead in the KEP thought it may be insignificant.